src/EFCore.PG/Extensions/NpgsqlIndexBuilderExtensions.cs

```diff
@@ -47,5 +52,25 @@ public static IndexBuilder ForNpgsqlHasOperators([NotNull] this IndexBuilder ind
 
             return indexBuilder;
         }
+
+        public static IndexBuilder ForNpgsqlInclude([NotNull] this IndexBuilder indexBuilder, [CanBeNull] params string[] propertyNames)
+        {
+            Check.NotNull(indexBuilder, nameof(indexBuilder));
+            Check.NullButNotEmpty(propertyNames, nameof(propertyNames));
+
+            indexBuilder.Metadata.Npgsql().IncludeProperties = propertyNames;
+
+            return indexBuilder;
+        }
+
+        public static IndexBuilder<TEntity> ForNpgsqlInclude<TEntity>([NotNull] this IndexBuilder<TEntity> indexBuilder, [NotNull] Expression<Func<TEntity, object>> includeExpression)
+        {
+            Check.NotNull(indexBuilder, nameof(indexBuilder));
+            Check.NotNull(includeExpression, nameof(includeExpression));
+
+            indexBuilder.ForNpgsqlInclude(includeExpression.GetPropertyAccessList().Select(x => x.Name).ToArray());
+
+            return indexBuilder;
```

Comment:
Can you return directly from the expression at line 71?

Reply:
Not really. The `ForNpgsqlInclude` call on line 71 returns `IndexBuilder` whereas this method requires `IndexBuilder<TEntity>`.

Reply:
Makes sense, thanks for the explanation!